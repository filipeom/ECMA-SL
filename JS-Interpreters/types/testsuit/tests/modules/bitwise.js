~ 5	      // retval = -6
5 & 1     // retval = 1
5 | 1	    // retval = 5
5 << 1	  // retval = 10
5 ^ 1	    // retval = 4
5 >> 1	  // retval = 2	
5 >>> 1	  // retval = 2



// retval = 2