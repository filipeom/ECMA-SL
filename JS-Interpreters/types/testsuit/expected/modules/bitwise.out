PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Program [stmts]:"
PROGRAM PRINT: ["$loc_3", "$loc_6", "$loc_10", "$loc_14", "$loc_18", "$loc_22", "$loc_26"]
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_4" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "prefix": true, "type": "UnaryExpression", "argument": "$loc_5", "operator": "~" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "BitwiseExpressionNot [arg]"
PROGRAM PRINT: { "type": "Literal", "raw": "5", "value": 5 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: -6.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_7" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "BinaryExpression", "right": "$loc_9", "left": "$loc_8", "operator": "&" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "BitwiseExpressionAnd [left][right]"
PROGRAM PRINT: { "type": "Literal", "raw": "5", "value": 5 }
PROGRAM PRINT: { "type": "Literal", "raw": "1", "value": 1 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 1.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 1.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_11" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "BinaryExpression", "right": "$loc_13", "left": "$loc_12", "operator": "|" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "BitwiseExpressionOr [left][right]"
PROGRAM PRINT: { "type": "Literal", "raw": "5", "value": 5 }
PROGRAM PRINT: { "type": "Literal", "raw": "1", "value": 1 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 1.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_15" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "BinaryExpression", "right": "$loc_17", "left": "$loc_16", "operator": "<<" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "BitwiseExpressionZeroFillLeftShift [left][right]"
PROGRAM PRINT: { "type": "Literal", "raw": "5", "value": 5 }
PROGRAM PRINT: { "type": "Literal", "raw": "1", "value": 1 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 1.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 10.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_19" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "BinaryExpression", "right": "$loc_21", "left": "$loc_20", "operator": "^" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "BitwiseExpressionXor [left][right]"
PROGRAM PRINT: { "type": "Literal", "raw": "5", "value": 5 }
PROGRAM PRINT: { "type": "Literal", "raw": "1", "value": 1 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 1.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 4.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_23" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "BinaryExpression", "right": "$loc_25", "left": "$loc_24", "operator": ">>" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "BitwiseExpressionSignedRightShift [left][right]"
PROGRAM PRINT: { "type": "Literal", "raw": "5", "value": 5 }
PROGRAM PRINT: { "type": "Literal", "raw": "1", "value": 1 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 1.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 2.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_27" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "BinaryExpression", "right": "$loc_29", "left": "$loc_28", "operator": ">>>" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "BitwiseExpressionZeroFillRightShift [left][right]"
PROGRAM PRINT: { "type": "Literal", "raw": "5", "value": 5 }
PROGRAM PRINT: { "type": "Literal", "raw": "1", "value": 1 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 1.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 2.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "End of the program [retval]:"
PROGRAM PRINT: 2.
